x-pack/plugins/ml/server/models/data_frame_analytics/analytics_manager.ts

```diff
@@ -243,31 +283,34 @@ export class AnalyticsManager {
     details[modelNodeId] = data;
     // fetch source job data and create elements
     if (sourceJobId !== undefined) {
-      data = await this.getAnalyticsData(sourceJobId);
-
-      nextLinkId = data?.source?.index[0];
-      nextType = JOB_MAP_NODE_TYPES.INDEX;
+      try {
+        data = await this.getAnalyticsData(sourceJobId);
+        nextLinkId = data?.source?.index[0];
+        nextType = JOB_MAP_NODE_TYPES.INDEX;
 
-      previousNodeId = `${data.id}-${JOB_MAP_NODE_TYPES.ANALYTICS}`;
+        previousNodeId = `${data.id}-${JOB_MAP_NODE_TYPES.ANALYTICS}`;
 
-      resultElements.push({
-        data: {
-          id: previousNodeId,
-          label: data.id,
-          type: JOB_MAP_NODE_TYPES.ANALYTICS,
-          analysisType: getAnalysisType(data?.analysis),
-        },
-      });
-      // Create edge between job and model
-      modelElements.push({
-        data: {
-          id: `${previousNodeId}~${modelNodeId}`,
-          source: previousNodeId,
-          target: modelNodeId,
-        },
-      });
+        resultElements.push({
+          data: {
+            id: previousNodeId,
+            label: data.id,
+            type: JOB_MAP_NODE_TYPES.ANALYTICS,
+            analysisType: getAnalysisType(data?.analysis),
+          },
+        });
+        // Create edge between job and model
+        modelElements.push({
+          data: {
+            id: `${previousNodeId}~${modelNodeId}`,
+            source: previousNodeId,
+            target: modelNodeId,
+          },
+        });
 
-      details[previousNodeId] = data;
+        details[previousNodeId] = data;
+      } catch (e) {
+        // fail silently if job doesn't exist
```

Comment:
I don't think it's a fair assumption that the error could only occur when the job doesn't exist. Worth checking the status code for 404, otherwise we should throw an error.

Reply:
Good point - checking explicitly for a 404 and throwing for other types of errors. Updated in 626ab62bb02d68546b25cf3cbf527a0fd88cc1a9

Reply:
nit https://github.com/elastic/kibana/pull/84299#discussion_r535926787